crates/api_common/src/utils.rs

```diff
@@ -276,31 +275,14 @@ async fn check_community_block(
   }
 }
 
-/// Throws an error if a recipient has blocked an instance.
 #[tracing::instrument(skip_all)]
-async fn check_instance_block(
-  instance_id: InstanceId,
-  person_id: PersonId,
-  pool: &mut DbPool<'_>,
-) -> Result<(), LemmyError> {
-  let is_blocked = InstanceBlock::read(pool, person_id, instance_id).await?;
-  if is_blocked {
-    Err(LemmyErrorType::InstanceIsBlocked)?
-  } else {
-    Ok(())
-  }
-}
-
-#[tracing::instrument(skip_all)]
-pub async fn check_person_instance_community_block(
+pub async fn check_person_community_block(
   my_id: PersonId,
   potential_blocker_id: PersonId,
-  instance_id: InstanceId,
   community_id: CommunityId,
   pool: &mut DbPool<'_>,
 ) -> Result<(), LemmyError> {
   check_person_block(my_id, potential_blocker_id, pool).await?;
-  check_instance_block(instance_id, potential_blocker_id, pool).await?;
   check_community_block(community_id, potential_blocker_id, pool).await?;
```

Comment:
Not entirely sure what the logic is here, shouldnt it be `check_community_block(community_id, my_id, pool)`? Would be good to have a comment explaining it.

Reply:
There's already a check community block just below this, to make sure they don't get sent to someone who's blocked that community.